src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -438,6 +446,30 @@ export default abstract class ComparisonStore {
                     this.molecularProfilesInActiveStudies.result!
                 )
             ),
+        onResult: profiles => {
+            if (!_.isEmpty(profiles)) {
```

Comment:
this kind of seems like it is defining default values?   i think we need more comment here in order to understand what this is doing.   also,   since we may be mutation state multiple times (forEach) we should run this whole in thing inside runInAction.   

but lets see if there's any way to do this without using onResult  would componentDidMount not work?